src/main/java/com/projectkorra/core/math/Angle.java

```diff
@@ -0,0 +1,75 @@
+package com.projectkorra.core.math;
+
+public class Angle {
+
+	private AngleMode mode;
+	private double value;
+
+	private Angle(AngleMode mode, double value) {
+		this.mode = mode;
+		this.value = value;
+	}
+
+	public AngleMode getMode() {
+		return mode;
+	}
+
+	public double getRawValue() {
+		return value;
+	}
+
+	public double getValue(AngleMode mode) {
+		if (this.mode != mode) {
+			return value * mode.getConversion();
+		} else {
+			return value;
+		}
+	}
+	
+	public void set(AngleMode mode, double value) {
+		this.mode = mode;
+		this.value = value;
+	}
+	
+	public void setValue(double value) {
+		this.value = value;
+	}
+
+	public static Angle radians(int value) {
+		return new Angle(AngleMode.RADIANS, (double) value);
+	}
+
+	public static Angle radians(float value) {
+		return new Angle(AngleMode.RADIANS, (double) value);
+	}
+
+	public static Angle radians(double value) {
+		return new Angle(AngleMode.RADIANS, value);
+	}
+
+	public static Angle degrees(int value) {
+		return new Angle(AngleMode.DEGREES, (double) value);
+	}
+
+	public static Angle degrees(float value) {
+		return new Angle(AngleMode.DEGREES, (double) value);
+	}
+
+	public static Angle degrees(double value) {
+		return new Angle(AngleMode.DEGREES, value);
+	}
```

Comment:
Java is able to implicitly cast `int` and `float` to `double` so I don't think you need anything other than the double typed methods.